File: backend/app/schemas/award.py
import uuid
from datetime import date, datetime
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class AwardParcelSummary(BaseModel):
    parcel_id: uuid.UUID
    khasra_number: str
    village_name: str
    acquired_area_sqm: Decimal
    compensation_assessment_id: uuid.UUID
    assessment_reference: Optional[str] = None
    assessed_amount_inr: Decimal
    owner_names: List[str] = []


class AwardCreate(BaseModel):
    project_id: uuid.UUID
    award_number: Optional[str] = None
    award_date: Optional[date] = None
    parcel_ids: List[uuid.UUID] = Field(..., min_length=1, description="Parcels with finalized compensation assessments to include in the award")
    remarks: Optional[str] = None


class AwardStatusUpdate(BaseModel):
    status: str = Field(..., description="Target status: UNDER_REVIEW, APPROVED, ISSUED, CHALLENGED, CLOSED")
    remarks: Optional[str] = None


class AwardSignRequest(BaseModel):
    """CALA Demo e-Sign / Approval Stamp request."""
    remarks: Optional[str] = "Demo statutory declaration and CALA approval applied."


class AwardListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    award_number: str
    project_id: uuid.UUID
    project_title: str
    project_code: str
    award_date: date
    total_parcels_count: int
    total_area_acres: Decimal
    total_award_amount_inr: Decimal
    cala_user_name: str
    status: str
    has_demo_esign: bool
    created_at: datetime


class AwardDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    award_number: str
    project_id: uuid.UUID
    project_title: str
    project_code: str
    district_name: str
    state_name: str
    award_date: date
    total_parcels_count: int
    total_area_acres: Decimal
    total_award_amount_inr: Decimal
    status: str
    cala_user_id: uuid.UUID
    cala_user_name: str
    cala_designation: Optional[str] = None
    digital_sign_hash: Optional[str] = None
    approval_stamp_label: str = "Demo e-Sign / Approval Stamp (Simulation)"
    approved_by_user_name: Optional[str] = None
    approval_date: Optional[datetime] = None
    remarks: Optional[str] = None
    parcels: List[AwardParcelSummary] = []
    total_disbursed_inr: Decimal = Decimal("0.0")
    remaining_inr: Decimal = Decimal("0.0")
    disbursement_percent: Decimal = Decimal("0.0")
    created_at: datetime
    updated_at: datetime

File: frontend/src/lib/types/award.ts
export interface AwardParcelSummary {
  parcel_id: string;
  khasra_number: string;
  village_name: string;
  acquired_area_sqm: number | string;
  compensation_assessment_id: string;
  assessment_reference?: string | null;
  assessed_amount_inr: number | string;
  owner_names: string[];
}

export interface AwardListItem {
  id: string;
  award_number: string;
  project_id: string;
  project_title: string;
  project_code: string;
  award_date: string;
  total_parcels_count: number;
  total_area_acres: number | string;
  total_award_amount_inr: number | string;
  cala_user_name: string;
  status: string;
  has_demo_esign: boolean;
  created_at: string;
}

export interface AwardDetail {
  id: string;
  award_number: string;
  project_id: string;
  project_title: string;
  project_code: string;
  district_name: string;
  state_name: string;
  award_date: string;
  total_parcels_count: number;
  total_area_acres: number | string;
  total_award_amount_inr: number | string;
  status: string;
  cala_user_id: string;
  cala_user_name: string;
  cala_designation?: string | null;
  digital_sign_hash?: string | null;
  approval_stamp_label: string;
  approved_by_user_name?: string | null;
  approval_date?: string | null;
  remarks?: string | null;
  parcels: AwardParcelSummary[];
  total_disbursed_inr: number | string;
  remaining_inr: number | string;
  disbursement_percent: number | string;
  created_at: string;
  updated_at: string;
}

export interface AwardFilterParams {
  project_id?: string;
  status?: string;
  search?: string;
  page?: number;
  page_size?: number;
}
